scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -495,6 +495,11 @@ def add_parse_arguments(parser = None):
                         before device handler open serial port and invoke runner.
                         """)
 
+    parser.add_argument("--flash-before", action="store_true", default=False,
+                        help="""perform flash operation before device handler
+                        open serial port and invoke runner.
```

Comment:
1) `invoke runner` looks confusing, may be it means 'test harness' ?
2) `flash-before` is mutually exclusive with `flash-with-test`, isn't it ?

Reply:
1. Is it better now?
2. Don't see why it should be mutually exclusive, can you elaborate? Thanks!

Reply:
`flash-with-test` extends the fixed flash timeout with the current test's timeout to allow both to run while the serial log is already open, whereas the proposed `flash-before` mode, as I understand, should await no more than the flash timeout before starting to read serial with the test timeout.